snap/hooks/configure

```diff
@@ -1,4 +1,19 @@
-#!/bin/bash -e
+#!/bin/bash
+set -ex
+
+ospassword=$(snapctl get ospassword)
+extgateway=$(snapctl get extgateway)
+dns=$(snapctl get dns)
+
+if [ -z "$ospassword" -o -z "$extgateway" -o -z "$dns" ]; then
+    echo "Missing required config value."
+    snapctl get microstack
+    exit 1
+fi
+
+snap-openstack setup # Write out templates
+
+source $SNAP_COMMON/etc/microstack.rc
```

Comment:
This, along with the code in the install hook, is the heart of the review. Here's where we set variables, call our new snap-openstack setup method, and source our new .rc. Most of the rest of the review involves juggling templates ...